docs/sources/userguide/plugins.md

```diff
@@ -0,0 +1,157 @@
+Docker Plugins MVP
+==================
+
+Plugins provide a mechanism for hooking into the Docker engine to extend its behavior.
+
+HTTP over a local UNIX socket is used to communicate between Docker and its plugins.
+
+Plugins are distributed as containers.
```

Comment:
I agree with this @shykes. Bind-mounting (which is how we'd propose to expose the Docker API) is already arguably part of the docker permissions interface (let the container mess with this subtree on the host filesystem). Does that mean that the plugin loader interface should have `-v`?

To be clear, the choice is:
* generic `-v` support for plugin loader.
* some sort of specific `--allow-docker-api` flag to the plugin loader which special-cases bind-mounting in the docker sock.

I think @progrium preferred the latter. I'm cool with either. If you strongly prefer the former @shykes I'm fine with that.

One data point: the prototypical containerized `flocker-zfs-agent` I'm working on currently has 5 bind-mounts. BTW, I was imagining that the flocker extension would start the `flocker-zfs-agent` with its requisite bind-mounts via the docker api when it started... but we really do need the extensions to be able to "do stuff". I don't think we have time to define an interface with granular permissions. And in fact arguably the interface with granular permissions should be a later version of the Docker API!